src/execution_plan/optimizations/traverse_order.c

```diff
@@ -306,6 +199,36 @@ static void _select_entry_point(QueryGraph *qg, AlgebraicExpression **ae, rax *f
 	}
 }
 
+// Insertion sort to order exps arrays by score descending.
+static void _sort_exp_array(AlgebraicExpression **exps, int *exp_scores, int exp_count) {
```

Comment:
IIRC, insertion sort is preferred for very small N (I think the C++ sort implementation switches to it at N=20 or something). This could be deleted if desired, though.